File: 02_Object_Oriented_Programming/02_Class_Relationships/05_Dependency/java/Printer.java
/**
 * Dependency — Printer class.
 *
 * Key concept: Document USES Printer temporarily (method parameter).
 * There is no permanent link — Printer is only needed during a specific operation.
 */
public class Printer {

    private String printerName;

    public Printer(String printerName) {
        this.printerName = printerName;
    }

    public String getPrinterName() {
        return printerName;
    }

    public void print(String message) {
        System.out.println("[" + printerName + "] Printing: " + message);
    }

    @Override
    public String toString() {
        return "Printer(" + printerName + ")";
    }
}

File: 02_Object_Oriented_Programming/02_Class_Relationships/05_Dependency/javascript/printer.js
/**
 * Dependency — Printer class in JavaScript.
 */

class Printer {
  #printerName;

  constructor(printerName) {
    this.#printerName = printerName;
  }

  get printerName() {
    return this.#printerName;
  }

  print(message) {
    console.log(`[${this.#printerName}] Printing: ${message}`);
  }

  toString() {
    return `Printer(${this.#printerName})`;
  }
}

module.exports = Printer;
